subprojects/launcher/src/test/groovy/org/gradle/tooling/internal/provider/FileSystemWatchingBuildActionRunnerTest.groovy

```diff
@@ -128,4 +135,57 @@ class FileSystemWatchingBuildActionRunnerTest extends Specification {
         def ex = thrown IllegalStateException
         ex.message == "Enabling file system watching via --watch-fs (or via the org.gradle.vfs.watch property) with --project-cache-dir also specified is not supported; remove either option to fix this problem"
     }
+
+    def "force-enables watching when #description"() {
+        _ * startParameter.watchFileSystemMode >> WatchMode.DEFAULT
+        _ * startParameter.projectCacheDir >> null
+        _ * startParameter.continuous >> isContinuousBuild
+        if (hasDeployment) {
+            deployments.add(Stub(Deployment))
+        }
+
+        when:
+        runner.run(buildAction, buildController)
+
+        then:
+        1 * watchingHandler.afterBuildStarted(WatchMode.ENABLED, _, _, buildOperationRunner) >> true
+
+        then:
+        1 * buildOperationProgressEventEmitter.emitNowForCurrent({ FileSystemWatchingSettingsFinalizedProgressDetails details -> details.enabled })
+
+        then:
+        1 * delegate.run(buildAction, buildController)
+
+        then:
+        1 * watchingHandler.beforeBuildFinished(WatchMode.ENABLED, _, _, buildOperationRunner, _)
+
+        then:
+        0 * _
+
+        where:
+        description               | hasDeployment | isContinuousBuild
+        "using continuous build"  | false         | true
+        "a deployment is running" | true          | false
```

Comment:
Should we also test that we don't force-enable watching if continuous mode is disabled and there are no deployments here, or are the other tests enough for that?

Reply:
The other test already cover these cases.